feat: 라우팅 및 레이아웃 연결

devjaeyoon opened the pull request:
### ✨ 이슈 번호

- #26 

### 📌 설명

지금까지 만든 페이지들의 라우팅을 진행하고 react-router-dom의 outlet을 이용하여 헤더 컴포넌트를 레이아웃과 연결하는 작업을 진행한 Pull request입니다.

### 📃 작업 사항

- [x] 지금까지 만든 페이지들의 라우팅을 진행한다.
- [x] 헤더 컴포넌트를 MainLayout에 연결한다.
- [x] 페이지별로 레이아웃과 컴포넌트가 자연스럽게 연결되어 있는지 확인하고 수정 작업을 진행한다.
- [x] 비모바일 환경 배경색 적용 
- [x] ToggleButton에 Button 컴포넌트 누락으로 인한 에러 수정
- [x] settingOptions 경로 오타 수정 

### 💡 참고 사항

라우팅 및 레이아웃 연결 작업을 진행하며 ToggleButton에 Button 컴포넌트 누락으로 인한 에러와 settingOptions 경로 오타로 인한 에러의 수정 작업을 함께 진행했습니다. 이 점 참고 부탁드립니다.

### 💭 리뷰 사항 반영

- [x] 제거해도 괜찮은 라우팅 분기 코드 제거

### ✅ Pull Request 체크 사항

- [x] 가장 최신 브랜치를 pull 하였습니다.
- [x] base 브랜치명을 확인하였습니다.
- [x] 코드 컨벤션을 모두 확인하였습니다.
- [x] 브랜치명을 확인하였습니다.

devjaeyoon committed:
feat: 채널 재생 및 검색 결과 페이지 라우트 추가

- /search-result 경로에 Search 페이지 라우트 추가
- /channel/:id 경로에 ChannelPlayer 페이지 라우트 추가
- 기존 라우터 구조에 새로운 경로를 통합하여 기능 확장

devjaeyoon committed:
chore: 높이를 차지하는 불필요한 div 제거

devjaeyoon committed:
feat: MainLayout에 Header 추가 및 비모바일 환경 배경색 적용

- MainLayout 컴포넌트에 Header 컴포넌트를 포함시켜 모든 페이지 상단에 표시
- 모바일이 아닌 환경에서는 중앙 정렬 및 보라색 배경이 적용되도록 스타일링 추가
- 콘텐츠는 최대 480px로 제한하여 모바일 중심 UI 유지

devjaeyoon committed:
fix: ToggleButton에서 Button 컴포넌트 누락으로 인한 에러 수정

- ToggleButton 컴포넌트 내에서 Button UI 컴포넌트 import가 누락되어 있었음
- @/components/ui/Button 경로로 Button을 import하여 오류 해결
- 이제 토글 버튼이 정상적으로 렌더링됨

devjaeyoon committed:
fix: settingOptions import 경로 오타 수정 (constant → constants)

- ChannelPlayer 컴포넌트에서 settingOptions를 불러오는 경로에 오타가 있어 import 오류 발생
- @/constant/settingOptions → @/constants/settingOptions로
- 경로를 올바르게 수정하여 정상 작동하도록 함

devjaeyoon committed:
fix: Header 레이아웃 패딩으로 수정하여 너비 이탈 문제 해결

- 기존  마진으로 인해 Header가 레이아웃 영역을 벗어나는 문제가 발생
-  패딩으로 대체하여 콘텐츠 너비 안에 안전하게 배치되도록 수정

devjaeyoon committed:
feat: 경로에 따라 Header 컴포넌트를 동적으로 렌더링하도록 구현

-  경로에서는 기본 MainHeader 표시
- 에서는 RadioSearchInput 표시
- 에서는 ReportHeader 표시
-  및 기타 경로에서는 검색 아이콘이 없는 MainHeader 표시

devjaeyoon committed:
refactor: 레이아웃 통합에 따라 Settings 페이지에서 중복된 Header 제거

- MainLayout에서 Header를 공통적으로 렌더링하므로,
- Settings 페이지에 별도로 추가했던 <MainHeader />를 제거함
- 중복 렌더링 방지 및 유지보수성을 개선

d0-neii commented on src/components/header/index.jsx:
```diff
@@ -0,0 +1,29 @@
+    return <ReportHeader />;
+  }
+
+  if (pathname === "/settings") {
```
`pathname === "/settings"` 조건과 그 아래  return이 중복되는 분기처럼 보입니다~
조건문 없이 `return <MainHeader showSearchIcon={false} />;`만 남기는 건 어떨까요??

devjaeyoon replied:
다은님의 의견도 맞습니다. 전 다른 개발자분이 제 코드를 참고할 때 이해하기 좀 더 수월할 수 있도록 "/settings"를 명시하는게 더 좋다 생각하는데 @faber-h 님은 어떻게 생각하시나요?

faber-h replied:
말씀하신 것처럼 특정 경로에 대해 명시적으로 분기 처리를 해두면 코드를 읽는 입장에서 쉽게 이해할 수 있다는 점 공감해요~
다만, 기능적으로는 조건문 없이 바로 `return <MainHeader showSearchIcon={false} />;`만 남겨도 동작엔 문제가 없고 코드도 간결해져서 좋을 것 같아요!

faber-h replied:
말씀하신 것처럼 특정 경로에 대해 명시적으로 분기 처리를 해두면 코드를 읽는 입장에서 쉽게 이해할 수 있다는 점 공감해요~
다만, 기능적으로는 조건문 없이 바로 return ;만 남겨도 동작엔 문제가 없고 코드도 간결해져서 좋을 것 같아요!

devjaeyoon replied:
알겠습니다! 그럼 수정하겠습니다~!

d0-neii commented on src/components/header/MainHeader.jsx:
```diff
@@ -11,7 +11,7 @@ const MainHeader = ({ showSearchIcon = true }) => {
 
   return (
-    <div className="mt-8 mb-8 ml-6 flex items-center justify-between">
+    <div className="flex items-center justify-between pt-8 pb-8 pl-6">
```
마진을 사용하셨던 것을 pt-8 pb-8 pl-6 패딩으로 변경하신 이유가 궁금합니다!
혹시 특정한 레이아웃 이슈나 의도하신 변화가 있어서 바꾸신 걸까요??

devjaeyoon replied:
마진을 패딩으로 수정한 이유는 div에서 마진은 외부 영역이기 때문에 레이아웃 기준 상 고려되지 않으면 튀어나오는 원인이 됩니다. 따라서, 자연스럽고 제한된 모바일 환경 UI를 구성하기 위해 마진을 패딩으로 수정했습니다.

faber-h replied:
말씀하신 것처럼 특정 경로에 대해 명시적으로 분기 처리를 해두면 코드를 읽는 입장에서 쉽게 이해할 수 있다는 점 공감해요~
다만, 기능적으로는 조건문 없이 바로 return <MainHeader showSearchIcon={false} />;만 남겨도 동작엔 문제가 없고 코드도 간결해져서 좋을 것 같아요!

devjaeyoon committed:
fix: 모바일 브라우저 높이 문제 대응을 위해 min-h-screen을 min-h-dvh로 변경

- 일부 모바일 브라우저에서 주소창 등 UI로 인해 min-h-screen이 전체 화면을 넘는 문제 발생
- 이를 해결하기 위해 min-h-dvh(동적 뷰포트 높이) 사용

faber-h reviewed (dismissed):
LGTM🔥

devjaeyoon committed:
refactor: /settings 경로에 대한 중복 조건 제거

- /settings 경로에 대한 조건은 기본 fallback 로직과 동일한 결과를 반환하므로 불필요한 중복
- 코드 간결성을 위해 해당 조건문 제거

faber-h reviewed (approved):
LGTM🔥

Eun0713 reviewed (approved):
LGTM💯
